lib/Basic/LangOptions.cpp

```diff
@@ -392,11 +393,19 @@ void LangOptions::setHasAtomicBitWidth(llvm::Triple triple) {
     setMaxAtomicBitWidth(128);
     break;
 
+  case llvm::Triple::ArchType::avr:
+    setMaxAtomicBitWidth(32);
+    break;    
+
   default:
     // Some exotic architectures may not support atomics at all. If that's the
     // case please update the switch with your flavor of arch. Otherwise assume
     // every arch supports at least word atomics.
 
+    if (triple.isArch16Bit()) {
+      setMaxAtomicBitWidth(16);
```

Comment:
I'm now confused by this addition being present together with the avr case above. Should we *only* keep this one?

Reply:
I suppose I was going for a "general 16 bit" and a specific AVR one but it's a bit pointless at this time so maybe I'll remove this one?